packages/pyodide-kernel/src/kernel.ts

```diff
@@ -244,6 +247,8 @@ export class PyodideKernel extends BaseKernel implements IKernel {
     await this.ready;
     const result = await this._remoteKernel.execute(content, this.parent);
     result.execution_count = this.executionCount;
+    await this._executed.promise;
+    this._executed = new PromiseDelegate<void>();
```

Comment:
I have a stupid question. There is always one PromiseDelegate. The execute request starts before interacting with the promise_delegate. If two execution requests are started almost at the same time, what identifies which promise to resolve once the message arrives?